Gems/Atom/RPI/Code/Source/RPI.Public/Shader/ShaderVariantAsyncLoader.cpp

```diff
@@ -19,6 +19,48 @@ namespace AZ
         AZ_CVAR(uint32_t, r_ShaderVariantAsyncLoader_ServiceLoopDelayOverride_ms, 0, nullptr, ConsoleFunctorFlags::Null,
             "Override the delay between iterations of checking for shader variant assets. 0 means use the default value (1000ms).");
 
+        static Data::AssetId GetShaderVariantAssetUuidFromShaderVariantTreeId(const Data::AssetId& shaderVariantTreeAssetId, const AZ::Name& supervariantName, ShaderVariantStableId stableId)
+        {
+            AZStd::string variantTreeRelativePath;
+            AZ::Data::AssetCatalogRequestBus::BroadcastResult(variantTreeRelativePath, &AZ::Data::AssetCatalogRequests::GetAssetPathById, shaderVariantTreeAssetId);
+            if (variantTreeRelativePath.empty())
+            {
+                return {};
+            }
+
+            AZStd::string shaderName; // Just the file name, no extension, no parent directory.
+            if (!AZ::StringFunc::Path::GetFileName(variantTreeRelativePath.c_str(), shaderName))
+            {
+                return {};
+            }
+
+            AZStd::string folderPath;
+            if (!AZ::StringFunc::Path::GetFolderPath(variantTreeRelativePath.c_str(), folderPath))
+            {
+                return {};
+            }
+
+            AZStd::string shaderVariantProductPath;
+
+            if (supervariantName.IsEmpty())
+            {
+                shaderVariantProductPath = AZStd::string::format("%s%s%s_%s_%u.%s",
+                    folderPath.c_str(), AZ_CORRECT_FILESYSTEM_SEPARATOR_STRING, shaderName.c_str(),
+                    RHI::Factory::Get().GetName().GetCStr(), stableId.GetIndex(), ShaderVariantAsset::Extension);
```

Comment:
suggestion: switch to Path library so client code isn't dealing with platform specific path separators.